contrib/src/main/java/com/datatorrent/contrib/parquet/ParquetFilePOJOReader.java

```diff
@@ -0,0 +1,318 @@
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, doubleVal);
+            break;
+
+          case BINARY:
+            pojoSetters.get(afi.fieldInfo.getPojoFieldExpression()).set(obj, group.getValueToString(fieldIndex, 0));
+            break;
+
+          default:
+            throw new ParquetEncodingException("Unsupported column type: " + primitiveTypeName);
+
+        }
+      } catch (InvalidRecordException e) {
+        logger.error("Field not found in schema {} ", e);
+      }
+    }
+    return obj;
+  }
+
+  @Override
+  protected void emit(Object tuple)
+  {
+    output.emit(tuple);
+  }
+
+  /**
+   * Sets the {@link FieldInfo}s. A {@link FieldInfo} maps a store column to a
+   * pojo field name.<br/>
+   * The value from fieldInfo.column is assigned to
+   * fieldInfo.pojoFieldExpression.
+   *
+   * @description $[].columnName name of the Output Field in POJO
+   * @description $[].pojoFieldExpression expression to get the respective field
+   *              from generic record
+   * @useSchema $[].pojoFieldExpression outputPort.fields[].name
+   */
+  public void setFieldInfos(List<FieldInfo> fieldInfos)
+  {
+    this.fieldInfos = fieldInfos;
+  }
+
+  /**
+   * Creates a map representing fieldName in POJO:field in Generic Record:Data
+   * type
+   * 
+   * @return List of FieldInfo
+   */
+  public List<FieldInfo> createFieldInfoMap(String str)
+  {
+    fieldInfos = new ArrayList<FieldInfo>();
+
+    StringTokenizer strtok = new StringTokenizer(str, RECORD_SEPARATOR);
+
+    while (strtok.hasMoreTokens()) {
+      String[] token = strtok.nextToken().split(FIELD_SEPARATOR);
+
+      fieldInfos.add(new FieldInfo(token[0], token[1], SupportType.valueOf(token[2])));
+    }
+
+    return fieldInfos;
+  }
+
+  /**
+   * Generates setter for given field and POJO class
+   * 
+   * @param klass
+   * @param inputFieldName
+   * @return Setter
+   * @throws NoSuchFieldException
+   * @throws SecurityException
+   */
+  private Setter generateSettersForField(Class<?> klass, String inputFieldName)
+      throws NoSuchFieldException, SecurityException
+  {
+    Field f = klass.getDeclaredField(inputFieldName);
+    Class c = ClassUtils.primitiveToWrapper(f.getType());
+    Setter classSetter = PojoUtils.createSetter(klass, inputFieldName, c);
+    return classSetter;
+  }
+
+  /**
+   * Returns groupToPOJOFieldsMapping
+   * 
+   * @return groupToPOJOFieldsMapping
+   */
+  public String getGroupToPOJOFieldsMapping()
+  {
+    return groupToPOJOFieldsMapping;
+  }
+
+  /**
+   * Sets groupToPOJOFieldsMapping
+   * 
+   * @param groupToPOJOFieldsMapping
+   */
+  public void setGroupToPOJOFieldsMapping(String groupToPOJOFieldsMapping)
+  {
+    this.groupToPOJOFieldsMapping = groupToPOJOFieldsMapping;
+  }
+
+  /**
+   * Class that maps fieldInfo to its getters or setters
+   */
+  protected static class ActiveFieldInfo
+  {
+    final FieldInfo fieldInfo;
+    Object setterOrGetter;
```

Comment:
setterOrGetter seems to be unused... Can this be removed.. And as fieldInfo is a single object in this class... Do we really need this class?

Also there is a List<FieldInfo> variable... How is it seperate from this one? If its meant to serve same purpose, can they be combined?